pkg/util/pidlimit.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2019 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+)
+
+const (
+	procCgroup    = "/proc/self/cgroup"
```

Comment:
@nixpanic question, `proc` and `sys` will be shared btw host and container for both cephfs and rbd?

Reply:
Not sure what you mean by "shared"... They are available in the container, but the contents can be slightly different from what you would see on the host. I think the provisioners do not use `hostPIDs: true`, so PID 1 will be the provisioner, not systemd/init.

Reply:
nevermind I was aksing shared/ bind-mounted from host to the container?

rbd plugin uses `hostPIDs: true` will this causes any issue https://github.com/ceph/ceph-csi/blob/master/deploy/rbd/kubernetes/v1.14%2B/csi-rbdplugin.yaml#L19

Reply:
No, there is no problem with the rbd plugin and `hostPIDs: true`. The PID limit that gets configured is for the container, and those limits are configured through cgroups. It is only needed to fetch the cgroup/pids name from `/proc` and then it PID limit can be reconfigured through `/sys/fs/cgroup/pids/..`. In case the executable is not running inside a container, it could adjust the PID limit for more processes on the host. In that case, just don't pass a `--pidlimit=...` option :-)